vita3k/motion/src/motion.cpp

```diff
@@ -22,8 +22,113 @@
 #include <ctrl/state.h>
 
 #include <SDL3/SDL_gamepad.h>
+#include <SDL3/SDL_sensor.h>
+
 #include <numbers>
 
+#ifdef __ANDROID__
+
+#include <SDL3/SDL_system.h>
+
+#include <jni.h>
+
+static bool is_device_landscape = false;
+
+static void init_device_orientation() {
+    JNIEnv *env = reinterpret_cast<JNIEnv *>(SDL_GetAndroidJNIEnv());
+    jobject activity = reinterpret_cast<jobject>(SDL_GetAndroidActivity());
+    jclass clazz(env->GetObjectClass(activity));
+
+    jmethodID method_id = env->GetMethodID(clazz, "isDefaultOrientationLandscape", "()Z");
+
+    is_device_landscape = env->CallBooleanMethod(activity, method_id);
+
+    // clean up the local references.
+    env->DeleteLocalRef(activity);
+    env->DeleteLocalRef(clazz);
+}
+
+#else
+constexpr bool is_device_landscape = true;
+#endif
+
+static void detect_device_motion_support(MotionState &state) {
+    state.device_accel_id = 0;
+    state.device_gyro_id = 0;
+
+    int num_sensors;
+    auto sensors_id = SDL_GetSensors(&num_sensors);
```

Comment:
sensors_id should be freed with [SDL_free](https://wiki.libsdl.org/SDL3/SDL_free)() when it is no longer needed.